packages/wp-now/src/encode-as-multipart.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Encodes the Express request with files into multipart/form-data request body.
+ * Adaptation of https://github.com/WordPress/wordpress-playground/blob/45bf16970867cc6f23738224dacf201905adcab6/packages/php-wasm/universal/src/lib/encode-as-multipart.ts
+ */
+export async function encodeAsMultipart(req: Express.Request) {
```

Comment:
Could you just passthrough the raw request bytes to PHP without re-encoding them as multipart?

Reply:
Good question. I presume we could pass the data using an `object` to let Playground encode it as multipart ([reference](https://github.com/WordPress/wordpress-playground/blob/0df453c11565c05c0c3c9e464459bf9c74677430/packages/php-wasm/universal/src/lib/encode-as-multipart.ts#L7-L47)). The only tricky part would convert the data from `files` property of the request to `File` class so it can be processed properly.

Reply:
@fluiddot the browser has already encoded the request – it should be possible to read the raw request body bytes and pass them, unchanged, to Playground. There's no need to create `File` instances or multipart-encode anything in the JavaScript code.

Reply:
I spent time reviewing this logic while addressing a Studio app crash, as similar logic is [applied there](https://github.com/Automattic/studio/commit/f9eed4c1bcc57eabfeff09303d26b230c190351b). 

From what I gather, by the time we reach the point to pass the request values onto PHP, the [`express-fileupload`](https://www.npmjs.com/package/express-fileupload) package has relocated any file data from `req.body` to `req.files`.

The file uploading logic was [added](https://github.com/WordPress/wordpress-playground/commit/69aa6c87343954a2f4e341718bfac1abb2ca15b9#diff-5ba071da369e7b99379e26a407462acb97304f8741c748de5deb696c1ecc0472R36) in https://github.com/WordPress/wordpress-playground/pull/188. My perception is that using a package like `express-fileupload` or [`multer`](https://www.npmjs.com/package/multer) is typical and required, as Express does not provide default support for `multipart/form-data`. @sejas does this align with your understanding as well? 

I attempted to merely pass long both `req.body` and `req.files` to PHP as the `body` attribute, but that ended with "out of bounds" errors for the `Uint8Array`, as the files are not in a format that Playground's [multipart encoding](https://github.com/WordPress/wordpress-playground/blob/0df453c11565c05c0c3c9e464459bf9c74677430/packages/php-wasm/universal/src/lib/encode-as-multipart.ts#L7-L47) would recognize currently (instance of `File`). 

I imagine we'd either need to retain this `File` creation/mutlipart-encode logic or expand Playground's multipart-encode logic to accept different file data structures.

Reply:
@dcalhoun here's how `@wp-playground/cli` supports uploads without any multipart parsing:

https://github.com/WordPress/wordpress-playground/blob/946dd7225cec1b5bebbe33072b246609d9e4524a/packages/playground/cli/src/server.ts#L28-L41

Reply:
Related bug: https://github.com/WordPress/playground-tools/issues/259

Reply:
I've tested [the approach](https://github.com/WordPress/wordpress-playground/blob/946dd7225cec1b5bebbe33072b246609d9e4524a/packages/playground/cli/src/server.ts#L28-L41) you shared @adamziel and seems to work if I disable the Express `fileUpload` middleware. I presume the middleware is modifying the request in a way that produces the `out of bounds` error that @dcalhoun described [here](https://github.com/WordPress/playground-tools/pull/203#discussion_r1597233159). I'll create a PR to update this.

Reply:
[Here's the PR](https://github.com/WordPress/playground-tools/pull/261) with the suggestion applied. Let me know if someone could take a look, thanks 🙇 !

cc @sejas @adamziel